snap/snapcraft.yaml

```diff
@@ -1,22 +1,20 @@
-name: tldr # you probably want to 'snapcraft register <name>'
-base: core22 # the base snap is the execution environment for this snap
-version: '3.2.0' # just for humans, typically '1.2+git' or '1.3.2'
-summary: tldr python client # 79 char long summary
+name: tldr
+base: core24
+version: '3.2.0'
```

Comment:
version: '3.3.0'

Reply:
isn't 3.3.0  the new release version number?

Reply:
This should be only bumped after creating the tag and release or else the build would fail (since Snap's build system searches for the tag in Git repo) [^1]. Then I will be able to bump the version for stable over there in Snapcraft dashboard.

I have mentioned the same in the PR description "After this PR's merge, I will create a new release and once it is done will bump the Snapcraft files version.".

----

If I bump it now then the CI will fail with this:

```sh
Run snapcore/action-build@v1
Building Snapcraft project in "."...
Installing Snapcraft plus dependencies
  Ensuring runner is in the lxd group...
  /usr/bin/sudo groupadd --force --system lxd
  /usr/bin/sudo usermod --append --groups lxd runner
  Installing LXD...
  /usr/bin/sudo snap refresh lxd
  snap "lxd" has no updates available
  Initialising LXD...
  /usr/bin/sudo lxd init --auto
  Installed docker related packages might interfere with LXD networking: 
  /usr/bin/sudo iptables -P FORWARD ACCEPT
  Installing Snapcraft...
  /usr/bin/sudo snap install --channel stable --classic snapcraft
  snapcraft [8](https://github.com/kbdharun/test-tldr-snap/actions/runs/9710738165/job/26802027824#step:3:9).2.12 from Canonical** installed
/usr/bin/sg lxd -c snapcraft -v
Starting snapcraft, version 8.2.12
Logging execution to '/home/runner/.local/state/snapcraft/log/snapcraft-20240628-0[9](https://github.com/kbdharun/test-tldr-snap/actions/runs/9710738165/job/26802027824#step:3:10)3819.026119.log'
Launching managed ubuntu 24.04 instance...
Creating new instance from remote
Creating new base instance from remote
Creating new instance from base instance
Starting instance
Starting snapcraft, version 8.2.12
Logging execution to '/tmp/snapcraft.log'
Starting snapcraft, version 8.2.12
Logging execution to '/tmp/snapcraft.log'
Initialising lifecycle
Installing build-packages
Installing build-snaps
Pulling tldr
:: + craftctl default
:: error:
:: Traceback (most recent call last):
::   File "/snap/snapcraft/current/lib/python3.[10](https://github.com/kbdharun/test-tldr-snap/actions/runs/9710738165/job/26802027824#step:3:11)/site-packages/craft_parts/ctl.py", line 111, in main
::     ret = CraftCtl.run(cmd, args)
::   File "/snap/snapcraft/current/lib/python3.10/site-packages/craft_parts/ctl.py", line 45, in run
::     _client(cmd, args)
::   File "/snap/snapcraft/current/lib/python3.10/site-packages/craft_parts/ctl.py", line 98, in _client
::     raise RuntimeError(message)
:: RuntimeError: Failed to pull source: command ['git', 'clone', '--recursive', '--branch', '3.3.0', 'https://github.com/tldr-pages/tldr-python-client.git', '/root/parts/tldr/src'] exited with code 128.
'override-pull' in part 'tldr' failed with code 1.
Recommended resolution: Review the scriptlet and make sure it's correct.
Failed to execute snapcraft in instance.
Full execution log: '/home/runner/.local/state/snapcraft/log/snapcraft-20240628-093819.026[11](https://github.com/kbdharun/test-tldr-snap/actions/runs/9710738165/job/26802027824#step:3:12)9.log'
Error: The process '/usr/bin/sg' failed with exit code 1
```

Reply:
@kbdharun thanks for your feedback ! :+1: 
so if I understand. We need to add the new tag in github after the build was done using the previous tag in the snap build?
And after that, as the tag will be changed, we will need to change to the new tag in the snap config file for the snap build to success?
So two commits are required at the end.